dask_jobqueue/remote_slurm.py

```diff
@@ -0,0 +1,114 @@
+                f"{self.api_url}jobs/submit", json={"script": script}
+            )
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            logger.exception("SLURMJob request failed.")
+            raise RuntimeError from e
+        else:
+            return response.json()
+
+    def _job_id_from_submit_output(self, out):
+        # out is the JSON output from _submit_job request.post
+        # See https://slurm.schedmd.com/rest_api.html#v0.0.36_job_submission_response
+        return out["job_id"]
+
+    async def close(self):
+        logger.debug("Stopping worker: %s job: %s", self.name, self.job_id)
+        # https://slurm.schedmd.com/rest_api.html#slurmctldCancelJob
+        try:
+            response = requests.delete(f"{self.api_url}job/{self.job_id}")
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            logger.exception("SLURMJob request failed.")
+            raise RuntimeError from e
+        else:
+            return response
+
+    @classmethod
+    def _close_job(cls, job_id):
+        # Overriding to please weakref in Job.start, not used.
+        pass
+
+
+class RemoteSLURMCluster(SLURMCluster):
+    __doc__ = """ Launch Dask on a SLURM cluster
+
+    Parameters
+    ----------
+    api_url: str
+        A url (ending in /) pointing to the SLURM REST API
+        See https://slurm.schedmd.com/rest_api.html
+    queue : str
+        Destination queue for each worker job. Passed to `#SBATCH -p` option.
+    project : str
+        Accounting string associated with each worker job. Passed to `#SBATCH -A` option.
+    {job}
+    {cluster}
+    walltime : str
+        Walltime for each worker job.
+    job_cpu : int
+        Number of cpu to book in SLURM, if None, defaults to worker `threads * processes`
+    job_mem : str
+        Amount of memory to request in SLURM. If None, defaults to worker
+        processes * memory
+    job_extra : list
+        List of other Slurm options, for example -j oe. Each option will be prepended with the #SBATCH prefix.
+
+    Examples
+    --------
+    >>> from dask_jobqueue import SLURMCluster
+    >>> cluster = SLURMCluster(
+    ...     queue='regular',
+    ...     project="myproj",
+    ...     cores=24,
+    ...     memory="500 GB"
+    ... )
+    >>> cluster.scale(jobs=10)  # ask for 10 jobs
+
+    >>> from dask.distributed import Client
+    >>> client = Client(cluster)
+
+    This also works with adaptive clusters.  This automatically launches and kill workers based on load.
+
+    >>> cluster.adapt(maximum_jobs=20)
+    """.format(
+        job=job_parameters, cluster=cluster_parameters
+    )
+
+    def __init__(self, *args, **kwargs):
+        if "api_url" not in kwargs:
+            raise ValueError("api_url is missing from RemoteSlurmCluster.")
+
+        if not kwargs["api_url"].endswith("/"):
+            raise ValueError("api_url on RemoteSlurmCluster should end with /.")
```

Comment:
The lack of `/` is likely going to be common among users. I was wondering if we could normalize the URL for the user instead of raising an error?

Reply:
You're right. I thought about using https://github.com/gruns/furl for this, quite fond of it. but it seemed overkill to add a library for 1 url.

We could also do;

```
if not kwargs["api_url"].endswith("/"):
    kwargs["api_url"] == f"{kwargs['api_url']}/"
```

?

Reply:
👍🏽 for the simple approach without a new dependency

Reply:
Added